R/hal.R

```
@@ -170,16 +208,19 @@ fit_hal.default <- function(X,
     )
   }
 
-  # If someone tries to pass (glmnet) standardize argument through "..." throw error.
-  # This is done because the HAL algorithm requires standardize = F for the variation norm interpretation to hold.
-  assertthat::assert_that(
-    !("standardize" %in% names(dot_args)),
-    msg = "hal9001 does not support the standardize argument."
-  )
+  # Throw error if `standardize` (glmnet) argument is passed through "..." .
+  # This is done because the HAL algorithm requires `standardize = FALSE` for
+  # the variation norm interpretation to hold.
+#   assertthat::assert_that(
+#     !("standardize" %in% names(dot_args)),
+#     msg = "hal9001 does not support the standardize argument."
+#   )
+  standardize <- FALSE
```

@Larsvanderlaan What was the reasoning behind adding `standardize` as an argument and then internally overriding the input? I agree that it should be `FALSE` as the assertion tried to force, but this lends itself to users getting unexpected output when they set `standardize = TRUE` (which they shouldn't do) and then having their choice silently discarded.

---

Hi Nima,
Yeah that was a stupid way to fix an issue. I did that initially so that haldensify did not error due to "standardize" being passed twice to glmnet inside the fit_hal function (once being through the ... arguments). After discussing it with Jeremy last week, we decided it would be better to just catch all the ... arguments and do what sl3 does with sl3:::call_with_args. I will work on getting this changed.

---

Thanks for confirming @Larsvanderlaan and for working on the change. I guess the idea would be to wrap the `glmnet` call in something like `call_with_args`, which should work fine I think, though I vaguely recall `glmnet` having some issues with not declaring `formals` in a convenient manner, which (again, off of memory, so might be wrong) caused issues with `call_with_args` in the past.

If there's a change needed for `haldensify`, please open up an issue to help track it over in that repo and I will make that change as time allows. For now, `haldensify` is pinned to the last commit of v0.2.8 of `hal9001` so it won't break from any of the functionality introduced by your spline HAL implementation in v0.3.0+ of this package.